apps/server/src/modules/user-login-migration/service/migration-check.service.ts

```diff
@@ -12,22 +12,39 @@ export class MigrationCheckService {
 		private readonly userLoginMigrationRepo: UserLoginMigrationRepo
 	) {}
 
-	async shouldUserMigrate(externalUserId: string, systemId: EntityId, officialSchoolNumber: string): Promise<boolean> {
+	public async shouldUserMigrate(
+		externalUserId: string,
+		systemId: EntityId,
+		officialSchoolNumber: string
+	): Promise<boolean> {
 		const school: LegacySchoolDo | null = await this.schoolService.getSchoolBySchoolNumber(officialSchoolNumber);
+		const userLoginMigration: UserLoginMigrationDO | null =
+			school && school.id ? await this.userLoginMigrationRepo.findBySchoolId(school.id) : null;
 
-		if (school && school.id) {
-			const userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationRepo.findBySchoolId(
-				school.id
-			);
+		if (!school || !userLoginMigration) {
```

Comment:
maybe you do this check first, then load the migration and then do a second check, if migration is null